overlord/state/task.go

```diff
@@ -208,6 +208,7 @@ func (t *Task) SetStatus(new Status) {
 	if chg != nil {
 		chg.taskStatusChanged(t, old, new)
 	}
+	return t.status != old
```

Comment:
sorry if I wasn't clear, this is not what I mentioned, what we care about is whether chg.taskStatusChanged changed the status of the Change, not just the task. Because Cleanup tasks can run only once a Change itself has reached ready state, as you can see from TaskRunner.clean.